interface/modules/custom_modules/oe-module-faxsms/src/Controller/ClickatellSMSClient.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Clickatell SMS Controller
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\FaxSMS\Controller;
+
+class ClickatellSMSClient extends AppDispatch
+{
+    private $appKey;
+    private $phone;
+
+    public function __construct()
+    {
+        if (empty($GLOBALS['oefax_enable_sms'] ?? null)) {
+            throw new \RuntimeException(xlt("Access denied! Module not enabled"));
+        }
+        parent::__construct();
+    }
+
+    /**
+     * @return array|mixed
+     */
+    public function getCredentials(): mixed
+    {
+        $credentials = appDispatch::getSetup();
+
+        $this->appKey = $credentials['appKey'];
+        $this->phone = $credentials['phone'];
+
+        return $credentials;
+    }
+
+    /**
+     * @return string
+     */
+    public function sendSMS(string $toPhone = '', string $subject = '', string $message = '', string $from = ''): string
+    {
+        // $subject and $from do not have valid/meaningful values passed in
```

Comment:
I was curious why there were parameters for subject and from, when there were not values passed in.  The notification script generates the `$db_sms_msg`, but doesn't populate those fields and [falls back to empty strings](https://github.com/openemr/openemr/blob/master/interface/modules/custom_modules/oe-module-faxsms/library/rc_sms_notification.php#L179-L183), instead.

Reply:
message comes from dialogs or passed in. $db_sms_msg is a canned template in setups.

Reply:
In the old scripts, `$db_sms_msg` was queried from the database.  In the new script, it instead is [generated in place in the script](https://github.com/openemr/openemr/blob/master/interface/modules/custom_modules/oe-module-faxsms/library/rc_sms_notification.php#L116-L118).  Since we never set values for `email_subject` or `email_sender`, when we try to pull out values for parameters to `sendSMS()` we'll always [fall back to the empty strings](https://github.com/openemr/openemr/blob/master/interface/modules/custom_modules/oe-module-faxsms/library/rc_sms_notification.php#L179-L183).

(Since an SMS doesn't have a separate subject I wouldn't expect to even have it as a parameter, unless it was to unify an interface with email. For emails, though, there aren't parameters for sender or subject, and instead those are [hardcoded or global variables](https://github.com/openemr/openemr/blob/master/interface/modules/custom_modules/oe-module-faxsms/src/Controller/EmailClient.php#L144-L148).  Not anything that needs to be covered for this PR, but something I noticed on the way as a fresh set of eyes.)

Reply:
If you don't need then don't use. I'm unsure why they're there and may go back to five years ago. Message/comment comes from send SMS dialog events. Some internal send events allow for sending to both email and SMS especially sending onetime tokens. Some dialogs were shared at one time to conserve on firing events.
Look at NotificationEventListener.php and interface/modules/custom_modules/oe-module-faxsms/openemr.bootstrap.php